timelock-impl/src/main/java/com/palantir/atlasdb/timelock/TimestampManagementResource.java

```diff
@@ -0,0 +1,101 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock;
+
+import com.palantir.conjure.java.undertow.annotations.Handle;
+import com.palantir.conjure.java.undertow.annotations.HttpMethod;
+import com.palantir.conjure.java.undertow.annotations.SerializerFactory;
+import com.palantir.conjure.java.undertow.lib.Endpoint;
+import com.palantir.conjure.java.undertow.lib.Serializer;
+import com.palantir.conjure.java.undertow.lib.TypeMarker;
+import com.palantir.conjure.java.undertow.lib.UndertowRuntime;
+import com.palantir.logsafe.Safe;
+import com.palantir.timestamp.TimestampManagementService;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.util.Headers;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import javax.annotation.CheckReturnValue;
+import javax.annotation.meta.When;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+@Path("/{namespace: (?!(tl|lw)/)[a-zA-Z0-9_-]+}")
+public class TimestampManagementResource {
+    private static final long SENTINEL_TIMESTAMP = Long.MIN_VALUE;
+    private static final String SENTINEL_TIMESTAMP_STRING =
+            SENTINEL_TIMESTAMP + ""; // can't use valueOf/toString because we need a compile time constant!
+
+    private final TimelockNamespaces namespaces;
+
+    public TimestampManagementResource(TimelockNamespaces namespaces) {
+        this.namespaces = namespaces;
+    }
+
+    // Timestamp management
+    @POST
+    @Path("timestamp-management/fast-forward")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Handle(method = HttpMethod.POST, path = "/{namespace}/timestamp-management/fast-forward")
+    public void fastForwardTimestamp(
+            @Safe @PathParam("namespace") @Handle.PathParam String namespace,
+            @Safe
+                    @QueryParam("currentTimestamp")
+                    @DefaultValue(SENTINEL_TIMESTAMP_STRING)
+                    @Handle.QueryParam(value = "currentTimestamp")
+                    Long currentTimestamp) {
+        long timestampToUse = currentTimestamp == null ? SENTINEL_TIMESTAMP : currentTimestamp;
+        getTimestampManagementService(namespace).fastForwardTimestamp(timestampToUse);
+    }
+
+    @GET
+    @Path("timestamp-management/ping")
+    @Produces(MediaType.TEXT_PLAIN)
+    @CheckReturnValue(when = When.NEVER)
+    @Handle(
+            method = HttpMethod.POST,
+            path = "/{namespace}/timestamp-management/fast-forward",
```

Comment:
this doesn't seem right?? Should be ping

Reply:
and should be get, too

Reply:
ugh. 🤡 

Yeah this is my bad - unfortunately this is only exercised by the undertow branch, and this endpoint is only used in the timelock migration...